internal/app/synchronizer/evaluator_client.go

```diff
@@ -134,7 +150,77 @@ func (ec *evaluatorClient) grpcEvaluate(ctx context.Context, proposals []*pb.Mat
 	return results, nil
 }
 
-func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) (results []*pb.Match, err error) {
-	// TODO: Implement a bidirectional HTTP streaming client in v0.7
-	return nil, nil
+func (ec *evaluatorClient) httpEvaluate(ctx context.Context, proposals []*pb.Match) ([]*pb.Match, error) {
+	reqr, reqw := io.Pipe()
+	proposalIDs := getMatchIds(proposals)
+	waitc := make(chan error)
+
+	go func() {
+		var m jsonpb.Marshaler
+		defer close(waitc)
+		defer func() {
+			sendErr := reqw.Close()
+			if sendErr != nil {
+				logger.Warning("failed to close response body read closer")
+			}
+		}()
+		for _, proposal := range proposals {
+			buf, sendErr := m.MarshalToString(&pb.EvaluateRequest{Match: proposal})
+			if sendErr != nil {
+				waitc <- status.Errorf(codes.FailedPrecondition, "failed to marshal proposal to string: %s", sendErr.Error())
```

Comment:
Why FailedPrecondition? do we need to grpcfy the error we got? if yes, FailedPrecondition may not be right? that error indicates that if the state of system changes, it may succeed - that may not be the case here: https://github.com/grpc/grpc-go/blob/master/codes/codes.go#L88

Reply:
Changed it to codes.Aborted.